src/libopensc/padding.c

```diff
@@ -184,6 +184,135 @@ sc_pkcs1_strip_02_padding(sc_context_t *ctx, const u8 *data, size_t len, u8 *out
 	LOG_FUNC_RETURN(ctx, len - n);
 }
 
+#ifdef ENABLE_OPENSSL
+static int mgf1(u8 *mask, size_t len, u8 *seed, size_t seedLen, const EVP_MD *dgst)
+{
+	int i;
+	size_t outlen = 0;
+	u8 cnt[4];
+	EVP_MD_CTX *md_ctx = NULL;
+	int mdlen;
+	u8 md[EVP_MAX_MD_SIZE];
+	int rv = 1;
+
+	if (!(md_ctx = EVP_MD_CTX_new()))
+		goto out;
+
+	mdlen = EVP_MD_size(dgst);
+	if (mdlen < 0)
+		goto out;
+
+	for (i = 0; outlen < len; i++) {
+		cnt[0] = (u8) ((i >> 24) & 255);
+		cnt[1] = (u8) ((i >> 16) & 255);
+		cnt[2] = (u8) ((i >> 8) & 255);
+		cnt[3] = (u8) ((i >> 0) & 255);
+		if (!EVP_DigestInit_ex(md_ctx, dgst, NULL)
+		    || !EVP_DigestUpdate(md_ctx, seed, seedLen)
+		    || !EVP_DigestUpdate(md_ctx, cnt, 4))
+			goto out;
+		if (outlen + mdlen <= len) {
+			if (!EVP_DigestFinal_ex(md_ctx, mask + outlen, NULL))
+				goto out;
+			outlen += mdlen;
+		} else {
+			if (!EVP_DigestFinal_ex(md_ctx, md, NULL))
+				goto out;
+			memcpy(mask + outlen, md, len - outlen);
+			outlen = len;
+		}
+	}
+	rv = 0;
+ out:
+	OPENSSL_cleanse(md, sizeof(md));
+	if (md_ctx)
+		EVP_MD_CTX_free(md_ctx);
+	return rv;
+}
+
+/* forward declarations */
+static const EVP_MD *mgf1_flag2md(unsigned int mgf1);
+static const EVP_MD *hash_flag2md(unsigned int hash);
+
+/* check/remove OAEP - RFC 8017 padding */
+int sc_pkcs1_strip_oaep_padding(sc_context_t *ctx, u8 *data, size_t len, unsigned long flags)
+{
+	size_t i,j;
+	size_t mdlen, dblen;
+	u8 seed[EVP_MAX_MD_SIZE];
+	const EVP_MD *mgf1_md, *hash_md;
+	u8 db[512];		/* up to RSA 4096 */
+	u8 label[EVP_MAX_MD_SIZE];
+	EVP_MD_CTX *md_ctx;
+	unsigned int hash_len = 0;
+
+	if (*data != 0)
```

Comment:
I think it's better to check `data != NULL && len > 0`. If you move this check behind L278, you'll get it almost for free.

Reply:
Thanks for the hint,  PR is already updated.